src/WebJobs.Script.WebHost/Standby/StandbyManager.cs

```diff
@@ -107,12 +116,29 @@ public async Task InitializeAsync()
                     File.WriteAllText(Path.Combine(functionPath, "run.csx"), content);
 
                     _logger.LogInformation($"StandbyMode placeholder function directory created");
+
+                    // start a background timer to identify when specialization happens
+                    // specialization usually happens via an http request (e.g. scale controller
+                    // ping) but this timer is started as well to handle cases where we
+                    // might not receive a request
+                    _specializationTimer = new Timer(OnSpecializationTimerTick, null, _specializationTimerInterval, _specializationTimerInterval);
                 }
                 finally
                 {
                     _semaphore.Release();
                 }
             }
         }
+
+        private void OnSpecializationTimerTick(object state)
+        {
+            if (!_environment.InStandbyMode)
+            {
+                SpecializeHostAsync().GetAwaiter().GetResult();
```

Comment:
I meant to come back and address this -- to prevent blocking, I was going to the task and chain continuation to dispose